From: meissner@suse.de
Subject: supply format arguments to gzprintf().

Index: zlib-1.2.7/zlib.h
===================================================================
--- zlib-1.2.7.orig/zlib.h	2012-05-03 06:12:35.000000000 +0200
+++ zlib-1.2.7/zlib.h	2012-09-27 13:12:58.187146312 +0200
@@ -1329,7 +1329,11 @@
    error.
 */
 
-ZEXTERN int ZEXPORTVA gzprintf Z_ARG((gzFile file, const char *format, ...));
+ZEXTERN int ZEXPORTVA gzprintf Z_ARG((gzFile file, const char *format, ...))
+#ifdef __GNUC__
+	__attribute__((__format__(__printf__,2,3)))
+#endif
+;
 /*
      Converts, formats, and writes the arguments to the compressed file under
    control of the format string, as in fprintf.  gzprintf returns the number of
